var/spack/repos/builtin/packages/qgis/package.py

```diff
@@ -185,6 +220,7 @@ def cmake_args(self):
                 "-DLIBZIP_INCLUDE_DIR=" + self.spec["libzip"].prefix.include,
                 "-DLIBZIP_CONF_INCLUDE_DIR=" + self.spec["libzip"].prefix.lib.libzip.include,
                 "-DGDAL_CONFIG_PREFER_PATH=" + self.spec["gdal"].prefix.bin,
+                "-DGDAL_LIBRARY=" + self.spec["gdal"].prefix + "/lib64/libgdal.so",
```

Comment:
here `libs[0]` didn't work since it points to `prefix/lib` as opposed to `prefix/lib64`.

Reply:
`libs[0]` points to the location where `libgdal.*` is found. That might be in lib or in lib64. On my laptop, it's `prefix/lib/libgdal.dylib`, so your above code clearly wouldn't work. The whole point of `libs` is to dynamically search for the library.

Reply:
I think this is now indirectly fixed by providing `-DGDAL_CONFIG`

Reply:
ultimately QGIS's FindGDAL.cmake must have a bug in it, which should be dealt with later.